addon/mixins/disposable.js

```diff
@@ -14,6 +17,8 @@ export default Mixin.create({
     this._super(...arguments);
 
     this._registeredDisposables = undefined;
+
+    this.willDestroy.patched = true;
```

Comment:
Due to how Ember does super wrapping, if this mixin is mixed into anything that _also_ implements a `willDestroy` this flag will be lost.  Consider this rough analogue (which is basically what [Ember does here](https://github.com/emberjs/ember.js/blob/7c93445601cb9c917fbb64e80fd85e2f6ead0dbd/packages/ember-utils/lib/super.js#L53-L68)):

```js
function _wrap(func, superFunc) {
  function superWrapper() {
    let orig = this._super;
    this._super = superFunc;
    let ret = func.apply(this, arguments);
    this._super = orig;
    return ret;
  }

  return superWrapper;
}
```

So as you can see, when Ember wraps any other `willDestroy`'s this flag on the func will be lost.

I'd suggest that instead we move to using a private branding on the mixin, that we can check for.

Perhaps, something like:

```js
const DO_CLEANUP_YO = typeof Symbol === 'function' ? Symbol('ember-lifeline cleanup') : Math.floor(Math.random() * new Date());

export default Mixin.create({
  [DO_CLEANUP_YO]: true,

  /* other stuff here */
})
```

Reply:
This did occur to me, and I wasn't too keep on keeping flags on the object as properties. I like the symbol approach.

Reply:
Wouldn't we do:

```js
export default Mixin.create({
  init() {
    this.[DO_CLEANUP_YO] = true,
  }
  /* other stuff here */
})
```
To set on the instance, not the prototype?

Reply:
NVM, worked this out offline.